compiler/optimizer/optimizer.go

```diff
@@ -322,3 +340,144 @@ func pushDown(trunk *dag.Trunk) {
 	}
 	trunk.Pushdown = filter
 }
+
+// newRangePruner returns a new predicate based on the input predicate pred
+// that when applied to an input value (i.e., "this") with fields from/to, returns
+// true if comparisons in pred against literal values can for certain rule out
+// that pred would be true for any value in the from/to range.  From/to are presumed
+// to be ordered according to the order o.  This is used to prune metadata objects
+// from a scan when we know the pool-key range of the object could not satisfy
+// the filter predicate of any of the values in the object.
+func newRangePruner(pred dag.Expr, fld field.Path, o order.Which) *dag.BinaryExpr {
+	lower := &dag.This{Kind: "This", Path: field.New("from")}
+	upper := &dag.This{Kind: "This", Path: field.New("to")}
+	if o == order.Desc {
+		lower, upper = upper, lower
+	}
+	e, _ := buildRangePruner(pred, fld, lower, upper, o)
+	return e
+}
+
+// buildRangePruner creates a DAG comparison expression that can evalaute whether
+// a Zed value adhering to the From/To pattern can be excluded from a scan because
```

Comment:
Nit: This is styled as "from/to" in the newRangePruner doc comment.
```suggestion
// a Zed value adhering to the from/to pattern can be excluded from a scan because
```